specify/specify6
# Update to v6.8.03

## Testing
[Most recent build](https://github.com/specify/specify6/actions/runs/4056531476/jobs/6981135987)

**BigDecimal** :
- Enter any value into any BigDecimal field
- Test Validation (enter `0`, negative numbers, and invalid characters `a[]{.`...
- Run a Query on BigDecimal fields and see if the results match display on forms
- Create a Report/Label using any BigDecimal fields (and test `formatBigDecimal` and `truncateBigDecimal` Scriptlets)
- Export BigDecimal fields and verify the results match Form/Query Results display
- Do additional light testing with Latitude and Longitude fields in `Locality`

**Carry Forward**:
- Test Carry Forward on the following tables:
  - AccessionCitation
  - CollectionObjectCitation
  - DNASequencingRunCitaiton
  - TaxonCitation
-  Ensure all expected values are stored correctly in the database and display on forms
  - More specifically, check if the information of the base table for each Citation is carry forwarded correctly

**BusinessRules**:
- Make sure an Agent of type Other/Group can be given addresses, and that this address can be queried on

**Plugins**
- Ensure the WorldWind plugin button is not visible when using MacOS

### Changes:
**BigDecimals**:
-  Strip down BigDecimals on form displays
- Improve validation for BigDecimals on forms (accept negative numbers and zero)
- Strip down BigDecimals in QueryResults
- DO NOT strip BigDecimals in Lables/Report (and add new `truncateBigDecmials` Scriptlet)
- Strip BigDecimals in Data Exports

**Carry Forward**:
Added support for the following tables:
- AccessionCitation
- CollectionObjectCitation
- DNASequencingRunCitation
- TaxonCitation

**Business Rules**:
- Allow Agents of type Other and Group to have addresses

**Plugins**:
- Disable the WorldWind plugin on MacOS
